SCons/Tool/linkCommon/linkCommmonTests.py

```diff
@@ -0,0 +1,75 @@
+# MIT License
+#
+# Copyright The SCons Foundation
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+Unit tests for linkCommon package
+"""
+
+import unittest
+
+from SCons.Environment import Environment
+
+
+class SharedLibraryTestCase(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_shlib_symlink_emitter(self):
+        """Test shlib_symlink_emitter() """
+        env = Environment(tools=['gnulink'])
+
+        target = env.SharedLibrary('lib', 'lib.c', SHLIBSUFFIX=".so")
+
+        target_name = str(target[0])
+        self.assertEqual(str(target_name), 'liblib.so', "Expected target 'liblib.so' != '%s'" % target_name)
```

Comment:
That's not platform-independent... (as you noted in the comment which says the expected target wasn't found on Windows).  Need to control the prefix as well, I assume.

Reply:
Good catch.. I love windows.. ;)